pkg/apis/direct.csi.min.io/v1alpha1/new-types.go

```diff
@@ -0,0 +1,80 @@
+// This file is part of MinIO Direct CSI
+// Copyright (c) 2020 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:defaulter-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Namespaced,shortName=directcsidrive,singular=directcsidrive
+// +kubebuilder:printcolumn:name="State",type="string",JSONPath=".status.currentState"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+
+type DirectCSIDrive struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	ID            string            `json:"ID"`
+	Name          string            `json:"name,omitempty"`
+	OwnerNode     nodeReference     `json:"ownerNode"`
+	TotalCapacity string            `json:"totalCapacity"`
+	FreeCapacity  string            `json:"freeCapacity"`
+	Volumes       []DirectCSIVolume `json:"volumes"`
+	BlockDevice   string            `json:"blockDevice"`
+	Formatted     bool              `json:"formatted"`
+	FileSystem    string            `json:"filesystem,omitempty"`
+	Status        driveStatus       `json:"driveStatus"`
+}
+
+type nodeReference struct {
```

Comment:
This should be exported if we're using it

- Have you looked into objectReference, it already has uuid and name https://github.com/kubernetes/api/blob/master/core/v1/types.go#L5226

- Both subfields should also be imported if we're using this